### PR TITLE
cmd/charm/charmcmd: Make sure the channel option is used in cmd when channel is set and WithChannel invoked.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/59/
Test PASSed.

File: cmd/charm/charmcmd/listresources.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"gopkg.in/juju/charm.v6-unstable"
 	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
 	"launchpad.net/gnuflag"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient"
```
Comment (jrwren) [inline cmd/charm/charmcmd/listresources.go:16]: sort these?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/60/
Test PASSed.

**Comment by jrwren** [comment]
:+1:  Curious, were the changes to revoke, set, show never made when channels were changed?

File: cmd/charm/charmcmd/cmd.go [hunk 2]
```diff
@@ -102,10 +102,7 @@ func New() commandWithPlugins {
 	chcmd.register(&showCommand{})
 	chcmd.register(&termsCommand{})
 	chcmd.register(&whoamiCommand{})
-	chcmd.register(&listResourcesCommand{
-		newCharmstoreClient: charmstoreClientAdapter(newCharmStoreClient),
-		formatTabular:       tabularFormatter,
-	})
+	chcmd.register(&listResourcesCommand{})
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd.go:105]: ahhh, silence.

File: cmd/charm/charmcmd/listresources.go [hunk 3]
```diff
@@ -86,47 +67,55 @@ func (c *listResourcesCommand) SetFlags(f *gnuflag.FlagSet) {
...
 	if err != nil {
 		return errgo.Notef(err, "could not retrieve resource information")
 	}
 
-	return c.Write(ctx, charmID2resources[c.charmID.String()])
+	return c.Write(ctx, charmID2resources[c.id.String()])
 }
 
 func parseArgs(auth string, args []string) (string, string, *charm.URL, error) {
 	if err := cmd.CheckEmpty(args); err == nil {
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/listresources.go]: if len(args) == 0 {

File: cmd/charm/charmcmd/listresources_test.go [hunk 4]
```diff
@@ -23,138 +17,87 @@ type listResourcesSuite struct {
...
+func (s *listResourcesSuite) TestListResource(c *gc.C) {
+	called := 0
+	dir := c.MkDir()
+	s.PatchValue(charmcmd.ListResources, func(csClient *csclient.Client, id *charm.URL) (map[string][]params.Resource, error) {
+		called++
+		c.Assert(csClient, gc.NotNil)
+		c.Assert(id, gc.DeepEquals, charm.MustParseURL("wordpress"))
 		return map[string][]params.Resource{
-			"cs:" + charmID: []params.Resource{
+			"cs:wordpress": []params.Resource{
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/listresources_test.go]: I think you can omit the "[]params.Resource" here.

File: cmd/charm/charmcmd/revoke.go [hunk 5]
```diff
@@ -112,7 +112,7 @@ func (c *revokeCommand) Run(ctxt *cmd.Context) error {
 	}
 	defer client.jar.Save()
 
-	if c.id.Revision == -1 {
+	if c.channel != "" {
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/revoke.go:115]: Technically this changes behaviour so there should be a test that needs to change too, but I'll let you off this time :)

**Comment by rogpeppe** [comment]
LGTM, nice to see everything get a bit simpler, thanks.

File: cmd/charm/charmcmd/revoke.go (hunk 5 above)
Comment (fabricematrat) [inline cmd/charm/charmcmd/revoke.go:115]: thanks :)

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/61/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit: thanks @rogpeppe && @jrwren

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**